feat(emailverification) : Verify Email

unadine opened the pull request:
## What does this PR do?
Email verification upon successful signup

## Description of Task to be completed?
User should be able to receive a verification link via email upon successful signup

## How should this be manually tested?
npm run test

## Any background context you want to provide?

## What are the relevant pivotal tracker stories?


## Screenshots (if appropriate)

## Questions:

James-nyagatare commented on src/middlewares/auth.js:
```diff
@@ -21,7 +21,7 @@ export default class Auth {
    * @return {object} object of payload or error
    */
-  static resetPassAuth(req, res, next) {
+  static reqParamsAuth(req, res, next) {
```
Good variable naming.

James-nyagatare reviewed (approved):
LGTM @unadine

James-nyagatare commented:
@unadine  squash the commits into one

harerakalex commented on src/controllers/userController.js:
```diff
@@ -132,20 +133,41 @@ export default class UserController {
+      if (!user) return Response.error(res, code.notFound, 'User does not exist!');
+      if (user.isVerified) return Response.error(res, code.conflict, 'User is already verified');
+      await UserService.updateUser({ isVerified: true }, { email });
+      return Response.success(res, code.ok, 'You have been verified!', { firstName, lastName });
```
This data `firstName and lastName` that you are sending in response is not that necessary, It will be great if you send only a message.

harerakalex reviewed (changes requested):
Great job @unadine however, consider the below comment.

Commit:
feat(verify email):Email verification

- Sent email to user upon successful registration
- Test using mocha and chai

[Finishes ##176153405]

Mucyo-Aime-Christian commented on src/tests/user.test.js:
```diff
@@ -86,4 +87,13 @@ describe('User related tests:', () => {
   });
+  it('should verify the email', async () => {
+    const res = await chai.request(app).patch(`/api/v1/users/verifyEmail/${emailtoken}`).send(mockdata.sendEmail);
+    console.log(res.status);
+    expect(res.status).to.be.equal(200);
+  });
```
please consider removing this unnecessary console

Mucyo-Aime-Christian commented on src/tests/user.test.js:
```diff
@@ -86,4 +87,13 @@ describe('User related tests:', () => {
+    expect(res.status).to.be.equal(200);
+  });
+  it('should not verify email with invalid token', async () => {
+    const res = await chai.request(app).patch(`/api/v1/users/verifyEmail/${mockdata.invalidToken}`).send(mockdata.sendEmail);
+    expect(res.status).to.be.equal(401);
+  });
```
how is this supposed to be 401 ??

unadine replied:
it is passing an invalid token

Mucyo-Aime-Christian reviewed (commented):
well done @unadine, however, mind to work on feedbacks i left

Mucyo-Aime-Christian reviewed (approved):
LGTM

harerakalex reviewed (approved):
LGTM
